src/github.com/getlantern/enproxy/conn_impl.go

```diff
@@ -38,34 +37,52 @@ func Dial(addr string, config *Config) (*Conn, error) {
 		return nil, fmt.Errorf("Unable to dial proxy: %s", err)
 	}
 
+	go c.processWrites()
+	go c.processReads()
 	go c.processRequests(proxyConn)
 
-	return c, nil
+	return idletiming.Conn(c, c.config.IdleTimeout, func() {
```

Comment:
enproxy.Conn used to handle its own idle timing.  Since we have an idletiming package that already handles this and is well tested, I'm letting it take care of that.